src/services/sync/strategies/WebUntis/WebUntisSchoolyearSyncer.js

```diff
@@ -0,0 +1,503 @@
+                        && givenEntry.duration === newEntry.duration
+                        && givenEntry.room === newEntry.room) {
+						givenEntry.count += 1;
+						entryFound = true;
+					}
+				}
+				if (!entryFound) {
+					times[subjectName].push(newEntry);
+					stats.times.count += 1;
+				}
+			}
+
+			for (const courseName in courses) {
+				if ({}.hasOwnProperty.call(courses, courseName)) {
+					const scCourse = courses[courseName];
+					const courseTimes = times[courseName];
+					// Update times, considered are events that occurs at least twice a year
+					scCourse.times = courseTimes.filter(entry => entry.count >= 2).map(entry => ({
+						weekday: entry.weekday,
+						startTime: entry.startTime,
+						duration: entry.duration,
+						eventId: undefined,
+						room: entry.room,
+					}));
+				}
+			}
+		}
+
+		return Promise.resolve();
+	}
+
+	async collectData(data, stats, school) {
+		assert.ok(this.dryrun);
+
+		const courseService = this.app.service('courses');
+		const classService = this.app.service('classes');
+
+		stats.classes = {};
+
+		/**
+         * Mapping:
+         *
+         * Schul-Cloud: class, course (per class), lesson
+         * WebUntis: class, subject
+         * German: Klasse, Kurs, Fach, Schulstunde
+         */
+
+		for (const klass of data.classes) {
+			/**
+             * Obtain classes
+             */
+			const className = klass.name;
+			this.logInfo(`Handle ${className}\n`);
+			const scClasses = await classService.find({ query: { name: className }, paginate: false });
+
+			let scClass = scClasses[0];
+			if (scClass === undefined) {
+				// Create Schul-Cloud class?
+				const newClass = {
+					name: className,
+					schoolId: school._id,
+					nameFormat: 'static',
+					year: school.currentYear,
+				};
+				scClass = newClass;
+				scClass._id = null;
+
+				stats.classes[className] = {
+					missing: true,
+					name: className,
+					courses: {},
+				};
+			} else {
+				stats.classes[className] = {
+					missing: true,
+					name: className,
+					courses: {},
+				};
+			}
+			const statsClass = stats.classes[className];
+
+			const times = {};
+			for (const timetableEntry of klass.timetable) {
+				/** Obtain courses for subjects:
+                 *
+                 * - class
+                 * - (teacher)
+                 * - time series
+                 * - room
+                 */
+				const subjectName = timetableEntry.subject;
+
+				let scCourse = statsClass.courses[subjectName];
+				if (scCourse === undefined) {
+					const courseName = `${subjectName} ${scClass.name}`;
+					this.logInfo(`Handle ${courseName}\n`);
+					const scCourses = await courseService.find({
+						query: {
+							name: courseName,
+							classIds: scClass._id,
+							schoolId: school._id,
+						},
+						paginate: false,
+					});
+
+					scCourse = scCourses[0];
+					if (scCourse === undefined) {
+						times[subjectName] = [];
+						
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Trailing spaces not allowed.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4061937)